burn-candle/src/ops/module.rs

```diff
@@ -1,24 +1,90 @@
-use burn_tensor::ops::{
-    ConvOptions, ConvTransposeOptions, MaxPool2dBackward, MaxPool2dWithIndices, ModuleOps,
+use burn_tensor::{
+    ops::{ConvOptions, ConvTransposeOptions, MaxPool2dBackward, MaxPool2dWithIndices, ModuleOps},
+    Shape,
 };
+use candle_core::ToUsize2;
 
 use crate::{
     element::{CandleElement, FloatCandleElement, IntCandleElement},
-    CandleBackend,
+    ops::base::reshape,
+    CandleBackend, CandleTensor,
 };
 
 use super::base::{FloatTensor, IntTensor};
 
 impl<F: FloatCandleElement, I: IntCandleElement> ModuleOps<CandleBackend<F, I>>
     for CandleBackend<F, I>
 {
+    fn conv1d(
+        x: FloatTensor<Self, 3>,
+        weight: FloatTensor<Self, 3>,
+        bias: Option<FloatTensor<Self, 1>>,
+        options: ConvOptions<1>,
+    ) -> FloatTensor<Self, 3> {
+        let conv = x
+            .tensor
+            .conv1d(
+                &weight.tensor,
+                options.padding[0],
+                options.stride[0],
+                options.dilation[0],
+                options.groups,
+            )
+            .unwrap();
+        CandleTensor::new(match bias {
+            Some(bias) => conv
+                .broadcast_add(&bias.tensor.unsqueeze(1).unwrap())
+                .unwrap(),
+            None => conv,
+        })
+    }
+
     fn conv2d(
         x: FloatTensor<Self, 4>,
         weight: FloatTensor<Self, 4>,
         bias: Option<FloatTensor<Self, 1>>,
         options: ConvOptions<2>,
     ) -> FloatTensor<Self, 4> {
-        todo!()
+        assert!(
+            options.dilation[0] == options.dilation[1]
+                && options.padding[0] == options.padding[1]
+                && options.stride[0] == options.stride[1],
+            "Candle does not support per dimension options in convolutions"
+        );
```

Comment:
This probably breaks a lot of tests.

Reply:
Yes, that's why we can't run our convolution test suite 😢